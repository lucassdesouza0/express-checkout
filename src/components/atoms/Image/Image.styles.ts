import styled from "@emotion/styled";

import mq from "utils/queries";

export const SImage = styled("img")`
  border-radius: 5px;
  border: 1px solid #eee;
  width: ${(props) => `${props.width}px`};
  height: ${(props) => `${props.height}px`};
`;
