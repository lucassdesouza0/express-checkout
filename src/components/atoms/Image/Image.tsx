import React from "react";

import { SImage } from "./Image.styles";

interface ImageProps {
  src: string;
  width: string;
  height: string;
}

const Image = ({ src, width = "230", height = "214" }: ImageProps) => {
  return <SImage src={src} width={width} height={height} data-testid="image" />;
};

export default Image;
