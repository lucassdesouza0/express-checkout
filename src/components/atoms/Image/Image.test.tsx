import React from "react";
import { screen, render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import testRender from "../../../tests/render";

import mock from "../../../tests/__mocks__/product";

import Image from "./Image";

describe("Button", () => {
  const imgUrl =
    "https://images.prismic.io/barkyn/153c4252-8084-49b8-9931-3ec426cdd859_premium_box.png?auto=compress,format";
  test("it should mount", () => {
    testRender(<Image src={imgUrl} />);

    const img = screen.getByRole("img");

    expect(img).toBeInTheDocument();
  });

  test("it should match snapshot", () => {
    const { asFragment } = render(<Image src={imgUrl} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
