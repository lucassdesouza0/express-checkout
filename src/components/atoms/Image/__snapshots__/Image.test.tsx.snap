// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it should match snapshot 1`] = `
<DocumentFragment>
  <img
    class="css-1ep1cpx-SImage e65m3ty0"
    data-testid="image"
    height="214"
    src="https://images.prismic.io/barkyn/153c4252-8084-49b8-9931-3ec426cdd859_premium_box.png?auto=compress,format"
    width="230"
  />
</DocumentFragment>
`;
